pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -498,6 +495,65 @@ func (r *ReconcileNetworkAddonsConfig) stopTrackingObjects() {
 	r.statusManager.SetFromPods()
 }
 
+func updateObjectsLabels(crLables map[string]string, objs []*unstructured.Unstructured) error {
+	var err error
+	for _, obj := range objs {
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		// Label objects with version of the operator they were created by
+		labels[cnaov1.SchemeGroupVersion.Group+"/version"] = operatorVersionLabel
+
+		labels[names.COMPONENT_LABEL_KEY] = names.COMPONENT_LABEL_DEFAULT_VALUE
+		labels[names.MANAGED_BY_LABEL_KEY] = names.MANAGED_BY_LABEL_DEFAULT_VALUE
+
+		labels = updateLabelsFromCR(labels, crLables)
+		if err != nil {
+			return err
+		}
+
+		err = updateTemplateLabels(obj, labels)
+		if err != nil {
+			return err
+		}
+
+		obj.SetLabels(labels)
```

Comment:
it may be more coherent to move this before the `err = updateTemplateLabels(obj, labels)`  line